### DISCUSSION TITLE
MBR with IDs generated from external search engine?

### DISCUSSION THREAD

**Discussion by mtrnka** [MannLabs/alphapept#525] [General]
I'm wondering if its possible to use alphapept functions for feature detection and match-between-runs with a table of spectral matches generated by a different search engine?  I would like to map my PSMs to alphapept generated features and then transfer to some other raw files, and would appreciate any pointers on how to implement this or whether it is more effort than it would be worth.  Thanks.

**Comment by straussmaximilian** [comment]
Hi,
yes, this should be possible. You can call custom steps of the pipeline.
I would suggest doing everything in a Jupyter notebook for prototyping. 

These would be the steps:

- Load default settings and specify the file paths
- Call file conversion & feature finding from the interface.py
- Load your PSMs, and save them as the first search in the alphapept hdf5 container
- Call downstream align and match from interface.py 

If it would help, I could prepare such a notebook with a basic code scaffold to help you get started. Let me know!

**Reply by mtrnka** [reply]
Hi, thanks so much for the reply.  I have questions with step3.  

How strict is the format of the ID table here?  I'm working with results from a different search engine and can map most of the columns in the csv id file fairly easily, but some would be more problematic and I'm hoping they could just be left as NA.  

I should mention that I am working with crosslinked peptides and so I ultimately need extra columns.  Also, I'm not sure if alphapept tries to re-calculate the monoistopic peptide mass from the sequence, so I don't know if I need to "fake" a linear peptide that has the correct composition to get this to work.

In terms of mapping the CSMs to features initially, the ID file has scan_no, raw_idx, feature_idx, and query_idx. Is it enough to just have the scan_no and feature_idx?  These are easiest to get.  raw_idx, seems to be the index of just the MS2 scans (eg, if I remove the precursor scans from the scan_no).  I'm having trouble figuring out how I get the query_idx. Is this the index in the "features" table?

It would be really helpful to see the jupyter notebook outline that you mentioned.  Thanks again for your help!

**Comment by mtrnka** [comment]
I think query_idx is confusing because there is column named this in features and one in the peptide report and they mean something different in each.

**Reply by straussmaximilian** [reply]
Apologies for the delay; I will upload the Notebook shortly.
Yes, query_idx can be confusing, you are probably looking for 'raw_idx'?

There is an overview for the different column names here:
https://mannlabs.github.io/alphapept/file_formats.html

**Reply by mtrnka** [reply]
Sorry for the slow reply.  I am still interested in seeing the notebook if you have time to do it.  Thanks for your help.

**Comment by straussmaximilian** [comment]
Hi,

Attached is a notebook with an example from MaxQuant - how this would, in principle, look like. This is not sanity-tested, so it could easily be that I made a plus instead of a minus somewhere :)

During implementation, I noticed some things that could be critical for such an approach:

- If we take identification from an external search engine, they likely have their own calibration. This would cause a mismatch between MS1 and MS2. With MaxQuant, I can undo this calibration. This is necessary as I need to map the MS1 features to MS2. However, MS1 masses are not being calibrated this way.
- There is an alignment step but this aligns across samples and not internally.

Hope this helps.
[Archive.zip](https://github.com/MannLabs/alphapept/files/10570749/Archive.zip)

**Reply by mtrnka** [reply]
Thanks so much. I'll start playing around with it this week.
